test/apps/iodemo/io_demo.cc

```diff
@@ -1475,8 +1511,18 @@ class DemoClient : public P2pDemoCommon {
 
             elapsed_time = curr_time - start_time;
             if (elapsed_time > _test_opts.client_timeout) {
-                LOG << "timeout waiting for " << (_num_sent - _num_completed)
-                    << " replies";
+                std::stringstream ss, list_uncomp_ss;
+                size_t uncomp_to_disconnect;
+
+                get_uncompleted_servers_info(list_uncomp_ss,
```

Comment:
maybe make this "log_outstanding_operations()" and it will just print a line for each uncompleted operation and on which connection it is waiting

Reply:
how does the full output look now?

Reply:
```
[1623442916.951931] [DEMO] timeout waiting for 18 replies on the following connections (0 connections to disconnect):
1. [UCX-connection 0xed8d60: #19 2.1.3.15:20007] has 7 uncompleted operations (read: 3050/3051; write: 2987/2993)
2. [UCX-connection 0xf73f50: #24 2.1.3.15:20006] has 11 uncompleted operations (read: 2992/2994; write: 3082/3091)
```